include/constants.h

```diff
@@ -0,0 +1,21 @@
+#ifndef constants

```

Comment:
this file doesn't seem necessary. `sizeof(ssm_word_t)` is more self-explanatory inline and does not take up that much more space, and the debug stuff should be defined on a per-compilation unit basis